File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/protocols/questionanswer/QuestionAnswerImplV1_0.java
package com.evernym.verity.sdk.protocols.questionanswer;

import com.evernym.verity.sdk.protocols.questionanswer.v1_0.QuestionAnswerV1_0;

/*
 * NON_VISIBLE
 *
 * This is an implementation of QuestionAnswerImplV1_0 but is not viable to user of Verity SDK. Created using the
 * static QuestionAnswer class
 */
class QuestionAnswerImplV1_0 extends AskCommonImpl implements QuestionAnswerV1_0 {

    QuestionAnswerImplV1_0(String forRelationship,
                                  String questionText,
                                  String questionDetail,
                                  String[] validResponses,
                                  Boolean signatureRequired) {
        super(forRelationship, questionText, questionDetail, validResponses, signatureRequired);
    }

    QuestionAnswerImplV1_0(String forRelationship,
                                  String threadId,
                                  String answer) {
        super(forRelationship, threadId, answer);
    }

    QuestionAnswerImplV1_0(String forRelationship,
                           String threadId) {
        super(forRelationship, threadId);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Protocols/QuestionAnswer/QuestionAnswerImplV1_0.cs
using System;
using System.Json;

namespace VeritySDK
{
    /// <summary>
    /// This is an implementation of QuestionAnswerImplV1_0 but is not viable to user of Verity SDK. Created using the static QuestionAnswer class
    /// </summary>
    public class QuestionAnswerImplV1_0 : QuestionAnswerV1_0
    {

        public QuestionAnswerImplV1_0(String forRelationship,
                                  String questionText,
                                  String questionDetail,
                                  String[] validResponses,
                                  Boolean signatureRequired) : base(forRelationship, questionText, questionDetail, validResponses, signatureRequired) { }

        public QuestionAnswerImplV1_0(String forRelationship,
                                      String threadId,
                                      String answer) : base(forRelationship, threadId, answer) { }

        public QuestionAnswerImplV1_0(String forRelationship,
                               String threadId) : base(forRelationship, threadId) { }

    }
}